jerry-core/vm/vm.c

```diff
@@ -154,55 +154,38 @@ vm_op_get_value (ecma_value_t object, /**< base object */
  *         if the property setting is unsuccessful
  */
 static ecma_value_t
-vm_op_set_value (ecma_value_t object, /**< base object */
+vm_op_set_value (ecma_value_t base, /**< base object */
                  ecma_value_t property, /**< property name */
                  ecma_value_t value, /**< ecma value */
                  bool is_strict) /**< strict mode */
 {
-  ecma_object_t * object_p;
+  ecma_value_t result = ECMA_VALUE_EMPTY;
+  ecma_object_t *object_p;
+  ecma_string_t *property_p;
 
-  if (JERRY_UNLIKELY (!ecma_is_value_object (object)))
+  if (JERRY_UNLIKELY (!ecma_is_value_object (base)
```

Comment:
I think this part is always true if the second part is true.

Reply:
Since this case is extremely unlikely, I added the first condition to reduce the number of checks when `base` is an object, which is more likely. I can remove it if you want.

Reply:
This looks like a very hoth path, and this is checked twice. What about moving the `ecma_is_value_prop_name` part before this, and combine this part with the rest?

Reply:
Unfortunatley the errors must be thrown in this order, the base object needs to be checked first, followed by the property name. The checks could be reduced by adding a completely separate code path for the non-object case and duplicating/factoring out the property name coercion. What do you think?

Reply:
This is a key function of vm, so in this case I would not mind the extra binary size.